src/Infrastructure/LeanCode.Logging/UserIdLogsCorrelationMiddleware.cs

```diff
@@ -0,0 +1,38 @@
+using System;
+using System.Security.Claims;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Serilog.Context;
+
+namespace LeanCode.Logging
+{
+    public static class UserIdLogsCorrelationMiddleware
+    {
+        public static IApplicationBuilder UseUserIdLogsCorrelation(
+            this IApplicationBuilder builder,
+            string? userIdClaim = null)
+        {
+            return builder.Use(async (httpCtx, next) =>
+            {
+                using var userId = UserId(httpCtx, userIdClaim);
+
+                await next();
+            });
+        }
+
+        private static IDisposable? UserId(HttpContext httpCtx, string? userIdClaim)
+        {
+            if (httpCtx?.User.Identity?.IsAuthenticated ?? false)
+            {
+                var userId = userIdClaim is not null ? httpCtx.User.FindFirstValue(userIdClaim) : httpCtx.User.Identity.Name;
```

Comment:
It would be better to assume `sub` as the claim name if `userIdClaim` is null IMO. As far as I know, `User.Identity.Name` might contain strange data.